io-unix.c

```diff
@@ -50,26 +50,51 @@ void
 read_or_mmap_xcf(const char *filename,const char *unzipper)
 {
   struct stat statbuf ;
-  
-  free_or_close_xcf() ;
 
-  if( strcmp(filename,"-") != 0 ) {
-    if( access(filename,R_OK) != 0 )
-      FatalGeneric(21,"!%s",filename);
-  }
+  free_or_close_xcf() ;
 
   if( !unzipper ) {
-    const char *pc ;
-    pc = filename + strlen(filename) ;
-    if( pc-filename > 2 && strcmp(pc-2,"gz") == 0 )
-      unzipper = "zcat" ;
-    else if ( pc-filename > 3 && strcmp(pc-3,"bz2") == 0 )
-      unzipper = "bzcat" ;
+    char bufmagic[2] ;
+    // 0x1f, 0x8b, 0x08; 8 for deflate
+    char gzmagic[2] = { 0x1f, 0x8b } ;
+    // 'B', 'Z', 'h'; h for huffman
+    char bzmagic[2] = { 'B', 'Z' } ;
+    // 0xfd, '7', 'z', 'X', 'Z'
+    char xzmagic[2] = { 0xfd, '7' } ;
```

Comment:
I believe xz magic bytes should be `{ 0xfd, '7', 'z', 'X', 'Z', 0 }` according to section 2.1.1.1 in [the specifications](https://tukaani.org/xz/xz-file-format.txt).